src/Microsoft.Restier.Core/Query/DefaultQueryHandler.cs

```diff
@@ -58,7 +58,7 @@ internal static class DefaultQueryHandler
 
             // execute query
             QueryResult result;
-            var executor = context.GetApiService<IQueryExecutor>();
+            var executor = context.GetApiContextService<IQueryExecutor>();
```

Comment:
The most problematic ones in my mind would be DbContext and ApiBase.
DbContext is scoped so it'll be created with every InvocationContext, I remember in submit pipeline there's a nested invocation to query pipeline, so they'll use different DbContext.
ApiBase is scoped so it'll be disposed with every InvocationContext. That means we'd better figure out a way to create an ApiBase for each InvocationContext.